plugins/woocommerce/src/Admin/PluginsHelper.php

```diff
@@ -970,6 +982,86 @@ function ( $sub ) {
 		);
 	}
 
+	/**
+	 * Get formatted notice information for missing subscription.
+	 *
+	 * @return array notice information.
+	 */
+	public static function get_missing_subscription_notice() {
+		if ( ! WC_Helper::is_site_connected() ) {
+			return array();
+		}
+
+		if ( ! self::$can_show_expiring_or_missing_subs_notice ) {
+			return array();
+		}
+
+		if ( ! self::should_show_notice( self::DISMISS_MISSING_SUBS_NOTICE ) ) {
+			return array();
+		}
+
+		$subscriptions         = WC_Helper::get_subscription_list_data();
+		$missing_subscriptions = array_filter(
+			$subscriptions,
+			function ( $sub ) {
+				return ( ! empty( $sub['local']['installed'] ) && empty( $sub['product_key'] ) );
+			},
+		);
+
+		// Remove WUM from missing subscriptions list.
+		$missing_subscriptions = array_filter(
+			$missing_subscriptions,
+			function ( $sub ) {
+				return 'woo-update-manager' !== $sub['zip_slug'];
+			}
+		);
+
+		if ( ! $missing_subscriptions ) {
+			return array();
+		}
+
+		$total_missing_subscriptions = count( $missing_subscriptions );
+
+		$notice_data = self::get_subscriptions_notice_data(
+			$subscriptions,
+			$missing_subscriptions,
+			$total_missing_subscriptions,
+			array(
+				/* translators: 1) product name */
+				'single_manage'           => __( 'You don\'t have a subscription for <strong>%1$s</strong>. Subscribe to receive updates and streamlined support.', 'woocommerce' ),
+				/* translators: 1) total expired subscriptions */
+				'different_subscriptions' => __( 'You don\'t have subscriptions for <strong>%1$s Woo extensions</strong>. Subscribe to receive updates and streamlined support.', 'woocommerce' ),
+			),
+			'missing',
+		);
+
+		$button_link = add_query_arg(
+			array(
+				'utm_source'   => 'pu',
+				'utm_campaign' => 'pu_in_apps_screen_purchase',
+			),
+			self::WOO_SUBSCRIPTION_PAGE_URL
+		);
```

Comment:
This should be the [same "subscribe" link used in-app](https://github.com/woocommerce/woocommerce/blob/904dfaa2ffe948d191010dbf74d799abbaf99875/plugins/woocommerce-admin/client/marketplace/utils/functions.tsx#L455-L459). Sending folks to wccom subscription page can be confusing, & doesn't help them subscribe to a product.

Reply:
Thanks @mehamasum changed in https://github.com/woocommerce/woocommerce/pull/51060/commits/72b43acb050d346d7716b82ac91628e3ebed8971, I also changed the link for expired subscriptions in https://github.com/woocommerce/woocommerce/pull/51060/commits/e22f170a04973fcb240ed6f2622d7f1531f6f9a6 and left expiring subscriptions alone as the CTA for that is auto-renew.

Reply:
I've also updated test cases above to reflect this change.